bigbluebutton-html5/imports/ui/components/settings/component.jsx

```diff
@@ -1,221 +1,158 @@
-        newIndex = 0;
-      } else {
-        newIndex = this.state.focusSubmenu - 1;
-      }
-
-      this.setState({ focusSubmenu: newIndex });
-      return;
-    }
-
-    // up arrow
-    if (event.keyCode === 38) {
-      if (this.state.focusSubmenu <= 0) {
-        this.setState({ focusSubmenu: this.submenus.length - 1 }, function () {
-          this.setFocus();
-        });
-      } else {
-        this.setState({ focusSubmenu: this.state.focusSubmenu - 1 }, function () {
-          this.setFocus();
-        });
-      }
-
-      return;
-    }
-
-    // down arrow
-    if (event.keyCode === 40) {
-      if (this.state.focusSubmenu >= this.submenus.length - 1) {
-        this.setState({ focusSubmenu: 0 }, function () {
-          this.setFocus();
-        });
-      } else {
-        this.setState({ focusSubmenu: this.state.focusSubmenu + 1 }, function () {
-          this.setFocus();
-        });
-      }
-
-      return;
-    }
-
-    // spacebar or enter
-    if (event.keyCode === 32 || event.keyCode === 13) {
-      this.setState({ activeSubmenu: this.state.focusSubmenu });
-      return;
-    }
+    this.handleSettingsApply = props.updateSettings;
+    this.handleUpdateSettings = this.handleUpdateSettings.bind(this);
+    this.handleSelectTab = this.handleSelectTab.bind(this);
   }
 
-  /* Keeps the focusSubmenu variable at the correct value when
-   * tabbing or shift-tabbing out of the submenu array
-   */
-  handleFocus(index) {
-    this.setState({ focusSubmenu: index });
-  }
+  setHtmlFontSize(size) {
+    document.getElementsByTagName('html')[0].style.fontSize = size;
+  };
 
   render() {
-
     return (
       <Modal
         title="Settings"
         confirm={{
           callback: (() => {
-            this.setState({ activeSubmenu: 0, focusSubmenu: 0 });
-            console.log('SHOULD APPLY SETTINGS CHANGES');
-            this.props.handleSaveFontState();
+            this.handleSettingsApply(this.state.current);
           }),
           label: 'Save',
           description: 'Saves the changes and close the settings menu',
         }}
         dismiss={{
           callback: (() => {
-            this.setState({ activeSubmenu: 0, focusSubmenu: 0 });
-            console.log('SHOULD DISCART SETTINGS CHANGES');
-            this.props.handleRevertFontState();
+            this.setHtmlFontSize(this.state.saved.application.fontSize);
           }),
           label: 'Cancel',
           description: 'Discart the changes and close the settings menu',
         }}>
-          {this.renderSettingOptions()}
+          {this.renderModalContent()}
       </Modal>
     );
   }
-};
 
-Settings.defaultProps = { title: 'Settings' };
+  handleUpdateSettings(key, newSettings) {
+    let settings = {
+      current: this.state.current,
+    };
+    settings.current[key] = newSettings;
+    this.setState(settings);
+  }
+
+  handleSelectTab(tab) {
+    this.setState({
+      selectedTab: tab,
+    });
+  }
```

Comment:
Yes, you can still use keys to navigate as it was before. It was all removed because now we are using [react-tabs](https://github.com/reactjs/react-tabs) which takes care of this.